src/xcode/ENA/ENA/Source/Scenes/Checkin/CheckInDetail/TraceLocationDetailViewModel.swift

```diff
@@ -0,0 +1,118 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import OpenCombine
+
+final class TraceLocationDetailViewModel {
+
+	// MARK: - Init
+	
+	init(_ traceLocation: TraceLocation) {
+		self.traceLocation = traceLocation
+		self.locationDescription = traceLocation.description
+		self.locationAddress = traceLocation.address
+		
+		if let defaultDuration = traceLocation.defaultCheckInLengthInMinutes {
+			self.selectedDurationInMinutes = defaultDuration
+		} else {
+			let eventDuration = Calendar.current.dateComponents(
+				[.minute],
+				from: traceLocation.startDate ?? Date(),
+				to: traceLocation.endDate ?? Date()
+			).minute
+			// the 0 should not be possible since we expect either the defaultCheckInLengthInMinutes or the start and end dates to be available always
+			self.selectedDurationInMinutes = eventDuration ?? 0
+		}
+	}
+	
+	// MARK: - Internal
+		
+	@OpenCombine.Published var pickerButtonTitle: String?
+
+	let locationDescription: String
+	let locationAddress: String
+	var selectedDurationInMinutes: Int
+	var shouldSaveToContactJournal = true
+	var traceLocationStatus: TraceLocationDateStatus? {
+		guard let startDate = traceLocation.startDate,
+			  let endDate = traceLocation.endDate else {
+			return nil
+		}
+		if startDate > Date() {
+			return .notStarted
+		} else if endDate < Date() {
+			return .ended
+		} else {
+			return .inProgress
+		}
+	}
+	
+	func pickerView(didSelectRow numberOfMinutes: Int) {
+		selectedDurationInMinutes = numberOfMinutes
+		let components = numberOfMinutes.quotientAndRemainder(dividingBy: 60)
+		let date = Calendar.current.date(bySettingHour: components.quotient, minute: components.remainder, second: 0, of: Date())
+		if let hour = formattedHourString(date) {
+			pickerButtonTitle = hour + " " + AppStrings.Checkins.Details.hoursShortVersion
```

Comment:
Please use a localizable String like "%@ Std." because we cannot assume that in every language the unit comes after the digits with one space.